### DISCUSSION TITLE
Controlled checkbox selection

### DISCUSSION THREAD

**Discussion by SourceCipher** [glideapps/glide-data-grid#776] [General]
How can we select the checkbox just by clicking on any cell?

We basically have a functionality which is row highlighting on click of any cell, but I cant find a way to do this with this table. I see the option to highlight the cells but thats not what we need.  We need to be able to basically select the row like this on any cell selection:

![image](https://github.com/glideapps/glide-data-grid/assets/351018/059a0b4a-e639-45d2-9e2e-242b8244d094)

**Comment by dellamonica** [comment]
This is an old thread, but other people might search for this, so here's an answer.

Use `onGridSelectionChange` to listen to cell selection, and update a state of type `GridSelection` accordingly.

```typescript
const [gridSelection, setGridSelection] = useState<GridSelection>({
    columns: CompactSelection.empty(),
    rows: CompactSelection.empty()
  })
// ...
<DataGrid
      gridSelection={gridSelection}
      onGridSelectionChange={(sel) => {
        if (sel.rows.length === 0 && sel.current?.cell !== undefined) {
          const [, row] = sel.current.cell
          setGridSelection({
            ...gridSelection,
            columns: CompactSelection.empty(),
            rows: CompactSelection.empty().add(row)
          })
        } else {
          setGridSelection(sel)
        }
      }}
...
```
